include/room_member.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+//
+
+#ifndef room_member_h
+#define room_member_h
+
+#include <deque>
+#include <string>
+
+#include "tunnler.h"
+
+#include "RakNetStatistics.h"
+#include "RakPeerInterface.h"
+
+class RoomMember {
+	public:
+
+	struct MemberInformation {
+		std::string nickname;
+		std::string gameName;
+		MacAddress mac_address;
+	};
+
+	enum class Status {
+
+		//Actual states
+		StateIdle, // Default state
+		StateError, // Some error [permissions to network device missing or something]
+		StateJoining, // This is also while the server didn't receive a setup packet / nickname from this player
+		StateJoined,
+    
+		// Reasons for connection loss
+		StateLostConnection,
+		StateKicked,
+    
+		// Reasons why connection was rejected
+		StateRoomFull,
+		StateRoomDestroyed,
+		StateWrongVersion,
+		StateWrongPassword,
+		StateBanned,
+		StatePasswordRequired,
+		StateNicknameAlreadyInUse,
+	};
+
+	struct ChatEntry {
+		std::string nickname;
+		std::string message;
+	};
+
+	RoomMember(MemberInformation member_information);
+	RoomMember(std::string nickname, std::string gameName, MacAddress mac_address);
+	~RoomMember();
+
+	Status getStatus();
+	MemberInformation getMemberInformation();
+
+	// Retrieves the new chat lines since last calling this
+	std::deque<ChatEntry> popChat();
+
+	// Retrieves the new wifi packets filtered by PacketType and MacAddress since last calling this
+	std::deque<WifiPacket> popWifiPackets(WifiPacket::PacketType type, const MacAddress& mac_address);
+
+	void sendChat(std::string message);
+	void sendWifiPacket(const WifiPacket& packet);
+
+	//Returns some stats about the connection to the room
+	std::string getsStats();
+
+	//Returns the latest ping to the room
+	int getPing();
+
+	//Must be called periodically to receive new data
+	void update();
+
+	void join(std::string nickname, std::string server = "127.0.0.1", uint16_t serverPort = 1234, uint16_t clientPort = 0);
+	void leave();
+	
+	private:
+	Status status;
+	MemberInformation member_information;
+
+	// Pointers to the interfaces of our server and client.
+	// Note we can easily have both in the same program
+	RakNet::RakPeerInterface *client;
+
+	static unsigned char GetPacketIdentifier(const RakNet::Packet *p);
```

Comment:
Raknet adds a header to each packet it sends. Our chat/beacon/data packets will al have a different identifier. This function just extracts the Identifier. It will be used inside Update to decide what will happen with the packet.